src/orange/surf/SurfaceClipper.hh

```diff
@@ -11,85 +11,66 @@
 #include "orange/BoundingBox.hh"
 #include "orange/OrangeTypes.hh"
 
+#include "SurfaceFwd.hh"
 #include "VariantSurface.hh"
-#include "detail/AllSurfaces.hh"
-#include "detail/SurfaceClipperImpl.hh"
 
 namespace celeritas
 {
 //---------------------------------------------------------------------------//
 /*!
- * Truncate a bounding box to its intersection with a surface interior.
+ * Truncate a bounding zone using a surface.
+ *
+ * This \em reduces the size of inner and outer bounding boxes to fit a
+ * surface. The \c interior bounding box is entirely \em inside the surface,
+ * and \c exterior is entirely \em outside. Axes that cannot be determined
+ * inside or out are left unchanged.
  *
  * Even though most quadric surfaces are infinite, their intersection with a
- * bounding box may be a smaller bounding box. This operation accelerates
- * "distance to in" calculations.
+ * bounding box may be a smaller bounding box. Accounting for the current
+ * bounding box's size when considering further truncation is *not yet
+ * implemented*.
+ *
+ * Shrinking bounding boxes will accelerate "distance to in" and "distance
+ * to out" calculations.
  */
 class SurfaceClipper
 {
   public:
-    // Construct with a reference to the bbox being clipped
-    explicit inline SurfaceClipper(BBox* bbox);
+    // Construct with interior and exterior bounding boxes
+    explicit SurfaceClipper(BBox* interior, BBox* exterior);
```

Comment:
Mostly because the bounding zone is a detail of orangeinp and I wasn't sure I wanted to spread it around... I think once the construction work is done we can look at reorganizing the files more sensibly (surface clipper should maybe go into an `orangeinp` construction detail?) and then simplify this.